network/stream/stream.go

```diff
@@ -612,29 +573,28 @@ func (r *Registry) clientHandleOfferedHashes(ctx context.Context, p *Peer, msg *
 	}
 
 	if err := p.Send(ctx, wantedHashesMsg); err != nil {
-		p.logger.Error("error sending wanted hashes", "err", err)
-		p.Drop("error sending wanted hashes")
-		return
+		return fmt.Errorf("error sending wanted hashes: %w", err)
 	}
 	if ctr == 0 {
 		// request the next range in case no chunks wanted
-		r.requestSubsequentRange(ctx, p, provider, w, msg.LastIndex)
-		return
+		return r.requestSubsequentRange(ctx, p, provider, w, msg.LastIndex)
+	}
+
+	if errc == nil {
```

Comment:
This is a very nice catch. But, nil errc should not be an error. In fact, read from nil channel blocks so the select bellow will wait for SyncBatchTimeout, which does not look correct. Should we just return nil in this case?

Reply:
I agree, I think we should just return nil

Reply:
the only case where `errc` is `nil` is when the `if` statement in line `559` is entered. in that case, the `if` statement in line `576` would be entered too, and the subsequent range would be requested. as a result, it is not possible that this `if` block will ever be triggered so it is safe to remove it IMO